### PR TITLE
[Rails 6] Initial dev

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Connected to #5159 

## Why was this needed?

Was reviewing what would be involved in upgrading and ended up fixing some of the low hanging issues. 

## What does this do?

Fixes deprecation warnings and broken specs.

## Implementation notes

Used the `rails_upgrade?` conditional to ensure this is backwards compatible.

### CODE REVIEW

File: app/models/public_body.rb [hunk 1]
```diff
@@ -403,7 +403,7 @@ def self.internal_admin_body
       else
         AlaveteliLocalization.
           with_locale(AlaveteliLocalization.default_locale) do
-          PublicBody.
+          default_scoped.
```
[the remark this answers is not in the document]
Comment (gbp) [reply app/models/public_body.rb:406]: Errm this is wrong... we have `EnforcedStyle: trailing`

**Comment by gbp** [comment]
This is green for the current Rails 5.2, but still red for Rails 6 (without any deprecation warnings) but doesn't look like much left - mostly due to #4896 

```
Finished in 15 minutes 37 seconds (files took 10.47 seconds to load)
6575 examples, 16 failures, 5 pending
```

I think this could be reviewed as is as its backward compatible.

File: spec/controllers/alaveteli_pro/batch_downloads_controller_spec.rb [hunk 2]
```diff
@@ -90,9 +90,15 @@ def show(id: '1', format: 'abc')
           end
 
           it 'returns content disposition' do
-            expect(response.header['Content-Disposition']).to(
-              eq 'attachment; filename="NAME"'
-            )
+            if rails_upgrade?
+              expect(response.header['Content-Disposition']).to(
+                eq 'attachment; filename="NAME"; filename*=UTF-8\'\'NAME'
```
Comment (garethrees) [inline spec/controllers/alaveteli_pro/batch_downloads_controller_spec.rb:95]: Are the backslashes here escaping the single quotes?

Might be clear to put this inside a `%q()` string

**Comment by garethrees** [comment]
Waiting on #5575 to get completed before merging this.

**Comment by gbp** [comment]
@garethrees could you review 9612bf7 too.

**Pull request merged**